src/wallet/api.rs

```diff
@@ -136,6 +136,16 @@ pub enum UTXOSpendInfo {
     },
 }
 
+type IncomingCoinsPair<'a> = (
+    Vec<(ListUnspentResultEntry, &'a IncomingSwapCoin)>,
+    Vec<(ListUnspentResultEntry, &'a OutgoingSwapCoin)>,
+);
+
+type OutgoingCoinsPair<'a> = (
+    Vec<(&'a IncomingSwapCoin, ListUnspentResultEntry)>,
+    Vec<(&'a OutgoingSwapCoin, ListUnspentResultEntry)>,
+);
```

Comment:
What happens if you don't use a reference here? 
```suggestion
type OutgoingCoinsPair<'a> = (
    Vec<(IncomingSwapCoin, ListUnspentResultEntry)>,
    Vec<(OutgoingSwapCoin, ListUnspentResultEntry)>,
);
```

Reply:
If we remove lifetimes and references, we get 10 errors in `api.rs` and `wallet.rs`
PFA the screenshot for errors:
![2023-11-13-23:14:54-screenshot](https://github.com/utxo-teleport/teleport-transactions/assets/38693805/c65c235a-4cde-4241-bd4b-571e2363001a)

Reply:
We also need lifetimes:
```bash
error[E0106]: missing lifetime specifier
   --> src/wallet/api.rs:145:10
    |
145 |     Vec<(&IncomingSwapCoin, ListUnspentResultEntry)>,
    |          ^ expected named lifetime parameter
    |
help: consider introducing a named lifetime parameter
    |
144 ~ type OutgoingCoinsPair<'a> = (
145 ~     Vec<(&'a IncomingSwapCoin, ListUnspentResultEntry)>,
```

Reply:
> If we remove lifetimes and references from OutgoingCoinsPair, we get 2 mismatched types errors in api.rs

This is because the function definitions aren't matching.. Remove lifetime requirements everywhere and it will work.. 

> note: expected struct `Vec<(IncomingSwapCoin, ListUnspentResultEntry)>` found struct `Vec<(&IncomingSwapCoin, ListUnspentResultEntry)>`

This means it wanted a ``Vec<(IncomingSwapCoin..)` and the code generated a ``Vec<(&IncomingSwapCoin..)`. Just make the iterator generate the first thing.

Reply:
Clone stuffs in the code if required..